packages/api/src/lib/problem-api.ts

```diff
@@ -0,0 +1,18 @@
+import axios, { AxiosError } from 'axios';
+
+export async function getProblemsBySection(sectionUri: string, courseId?: string) {
+  const resp = await axios.get(`/api/get-problems-by-section`, {
+    params: {
+      sectionUri,
+      ...(courseId ? { courseId } : {}),
+    },
+  });
+  return resp.data;
+}
+
+export async function getProblemCountsByCourse(courseId: string) {
```

Comment:
keep the api name and spec function name aligned
```suggestion
export async function getCourseProblemCounts(courseId: string) {
```